pkg/agent/util/net_windows.go

```diff
@@ -32,36 +33,83 @@ import (
 )
 
 const (
-	ContainerVNICPrefix = "vEthernet"
-	HNSNetworkType      = "Transparent"
-	LocalHNSNetwork     = "antrea-hnsnetwork"
-	OVSExtensionID      = "583CC151-73EC-4A6A-8B47-578297AD7623"
-	namedPipePrefix     = `\\.\pipe\`
+	ContainerVNICPrefix  = "vEthernet"
+	HNSNetworkType       = "Transparent"
+	LocalHNSNetwork      = "antrea-hnsnetwork"
+	OVSExtensionID       = "583CC151-73EC-4A6A-8B47-578297AD7623"
+	namedPipePrefix      = `\\.\pipe\`
+	commandMaxRetry      = 5
+	commandRetryInternal = time.Second
 )
 
 func GetNSPath(containerNetNS string) (string, error) {
 	return containerNetNS, nil
 }
 
+func GetHostInterfaceStatus(ifaceName string) (string, error) {
+	cmd := fmt.Sprintf(`Get-NetAdapter -InterfaceAlias "%s" | Select-Object -Property Status | Format-Table -HideTableHeaders`, ifaceName)
+	out, err := CallPSCommand(cmd)
+	if err != nil {
+		return "", err
+	}
+	return strings.TrimSpace(out), nil
+}
+
+func retryOperator(operation string, exec func() error) error {
+	var err error
+	for i := 0; i < commandMaxRetry; i++ {
+		err = exec()
+		if err == nil {
+			return nil
+		}
+		time.Sleep(commandRetryInternal)
+	}
+	return fmt.Errorf("failed to %s after %d retries: %v", operation, commandMaxRetry, err)
+}
+
 // EnableHostInterface sets the specified interface status as UP.
 func EnableHostInterface(ifaceName string) error {
-	cmd := fmt.Sprintf("Enable-NetAdapter -InterfaceAlias %s", ifaceName)
-	return InvokePSCommand(cmd)
+	cmd := fmt.Sprintf(`Enable-NetAdapter -InterfaceAlias "%s"`, ifaceName)
+	// Enable-NetAdapter is not a block operation by our test. It returns immediately no matter if the interface
+	// has been enabled. So we need to check interface status to ensure the interface is up before return.
+	return retryOperator(cmd, func() error {
+		if err := InvokePSCommand(cmd); err != nil {
+			return err
+		}
+		status, err := GetHostInterfaceStatus(ifaceName)
+		if err != nil {
+			return err
+		}
+		if !strings.EqualFold(status, "Up") {
+			return fmt.Errorf("host interface %s is not up", ifaceName)
+		}
+		return err
+	})
 }
 
 // ConfigureInterfaceAddress adds IPAddress on the specified interface.
 func ConfigureInterfaceAddress(ifaceName string, ipConfig *net.IPNet) error {
 	ipStr := strings.Split(ipConfig.String(), "/")
-	cmd := fmt.Sprintf("New-NetIPAddress -InterfaceAlias %s -IPAddress %s -PrefixLength %s", ifaceName, ipStr[0], ipStr[1])
-	return InvokePSCommand(cmd)
+	cmd := fmt.Sprintf(`New-NetIPAddress -InterfaceAlias "%s" -IPAddress %s -PrefixLength %s`, ifaceName, ipStr[0], ipStr[1])
+	err := InvokePSCommand(cmd)
+	// If the addrress already exists, ignore the error.
+	if err == nil || strings.Contains(err.Error(), "already exists") {
```

Comment:
nit: process error first and make the normal path indent least.
```
if err != nil && !strings.Contains(err.Error(), "already exists") {
    return err
}
return nil
```

Reply:
Done.